lib/Mojo/Message/Request.pm

```diff
@@ -153,15 +153,23 @@ sub _parse_env {
   my $headers = $self->headers;
   my $url     = $self->url;
   my $base    = $url->base;
+
+  my $port = $env->{SERVER_PORT};
+
   for my $name (keys %$env) {
     my $value = $env->{$name};
     next unless $name =~ s/^HTTP_//i;
     $name =~ y/_/-/;
     $headers->header($name => $value);
 
     # Host/Port
-    $value =~ s/:(\d+)$// ? $base->host($value)->port($1) : $base->host($value) if $name eq 'HOST';
+    if ($name eq 'HOST') {
```

Comment:
Would this be easier to understand as `if ( not defined $port and $name eq 'HOST' )`

Reply:
@rmzg No, because that wouldn't set `$base->host` if `defined $port`